pkg/catalog/types_with_details_test.go

```diff
@@ -0,0 +1,222 @@
+		wantErr bool
+	}{
+		{
+			name: "get a pos 0",
+			types: TypesWithDetails{
+				"second devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile2",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+				},
+				"first devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry2",
+						},
+					},
+				},
+			},
+			args: args{
+				pos: 0,
+			},
+			want: DevfileComponentType{
+				Name: "devfile1",
+				Registry: Registry{
+					Name: "Registry1",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "get a pos 1",
+			types: TypesWithDetails{
+				"second devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile2",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+				},
+				"first devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry2",
+						},
+					},
+				},
+			},
+			args: args{
+				pos: 1,
+			},
+			want: DevfileComponentType{
+				Name: "devfile1",
+				Registry: Registry{
+					Name: "Registry2",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "get a pos 2",
+			types: TypesWithDetails{
+				"second devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile2",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+				},
+				"first devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry2",
+						},
+					},
+				},
+			},
+			args: args{
+				pos: 2,
+			},
+			want: DevfileComponentType{
+				Name: "devfile2",
+				Registry: Registry{
+					Name: "Registry1",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "get a pos 4: not found",
+			types: TypesWithDetails{
+				"second devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile2",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+				},
+				"first devfile for lang1": []DevfileComponentType{
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry1",
+						},
+					},
+					{
+						Name: "devfile1",
+						Registry: Registry{
+							Name: "Registry2",
+						},
+					},
+				},
+			},
+			args: args{
+				pos: 4,
+			},
+			want:    DevfileComponentType{},
+			wantErr: true,
+		}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.types.GetAtOrderedPosition(tt.args.pos)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("TypesWithDetails.GetAtOrderedPosition() error = %v, wantErr %v", err, tt.wantErr)
+				return
```

Comment:
Yes, I don't want to test the result is the test on error is already not ok. Note that this will return from the anonymous function (passed to `t.Run`), not the entire test function